### ISSUE TITLE
Using confluent-kafka-go example on windows with latest Gogland 

### ISSUE THREAD

**Issue by lghinet** [confluentinc/confluent-kafka-go#97]
hello 
i get the following error when i try to build the producer example , because as far as i know there is no pkg_config on windows .. 

```
GOROOT=C:\Go
GOPATH=C:/GoglandProjects
C:\Go\bin\go.exe get -t github.com/confluentinc/confluent-kafka-go/kafka/...
# pkg-config --cflags rdkafka
pkg-config: exec: "pkg-config": executable file not found in %PATH%
# pkg-config --cflags --static rdkafka
pkg-config: exec: "pkg-config": executable file not found in %PATH%
```

thank you

**Comment by edenhill** [comment]
Windows is currently not a supported platform.
It is not clear how the native dependencies (librdkafka, openssl, zlib, etc) are supposed to be managed on Windows.

Your best bet at this point is to download the [librdkafka.redist](https://www.nuget.org/packages/librdkafka.redist) NuGet package and extract the required files (nuget packages are zip files with a .nupkg extension).

**Comment by lghinet** [comment]
thanks for the fast reply 

so can i somehow import them (.dll or .nupkg) in go ?
ultimately my go app will be deployed in docker linux container

**Comment by edenhill** [comment]
If you are going to deploy on Linux I really wouldn't go through the (yet unknown) hassle of developing on Windows, unless you are up for the challenge - in which case I would be very excited to know how you sort it out :)

**Comment by lghinet** [comment]
From librdkafka.dll I have created a .def and .a file using gendef and dlltool from mingw and then changing .go files by adding 
`#cgo windows LDFLAGS: -L./librdkafka -lrdkafka`
but still not working i get 
```
>go build -tags static
.\00version.go:27:32: fatal error: librdkafka/rdkafka.h: No such file or directory
compilation terminated.
```

**Comment by edenhill** [comment]
rdkafka.h is included in the NuGet package as well.

**Comment by ttallskog** [comment]
DId you ever find a solution this problem? I am having similar problems...

**Comment by noemi-dresden** [comment]
I am having the same problem as well, did you figure out the solution?

**Comment by mcgradycchen** [comment]
@noemi-dresden @ttallskog 
Follow the steps https://github.com/confluentinc/confluent-kafka-go/issues/128#issuecomment-386041003

* Fix error about sys/socket.h

         In file included from src\vendor\github.com\confluentinc\confluent-kafka-go\kafka\00version.go:25:0:
         /usr/include/librdkafka/rdkafka.h:83:10: fatal error: sys/socket.h: No such file or directory
         #include <sys/socket.h> /* for sockaddr, .. */
      
* Modify the included header file `librdkafka/rdkafka.h`, add win macro define
   
         #define _MSC_VER 1900 //for win10
         #ifdef _MSC_VER

* Run go build again, then win the success

**Comment by oswee** [comment]
Didn't remember well, but related to socket.h issue i copied file `\github.com\username\project\librdkafka.redist.0.11.4-RC1B\runtimes\linux-x64\native\librdkafka.so` into `C:\cygwin64\lib`. And it fixed issue for me.

**Comment by spiderman0510** [comment]
confluent-kafka-go does not support Windows anymore?

**Comment by spiderman0510** [comment]
Hi @edenhill confluent-kafka-go does not support Windows anymore?

**Comment by edenhill** [comment]
There has never been official Windows support, but members of the community have been able to make it work on Windows.

**Comment by spiderman0510** [comment]
@edenhill , I did follow solutions from some guys in this issue https://github.com/confluentinc/confluent-kafka-go/issues/128, But got same error with this guy (https://github.com/confluentinc/confluent-kafka-go/issues/128#issuecomment-364943363). Do you have any idea about this issue?

**Comment by oswee** [comment]
@spiderman0510 While i made my life-time switch to Fedora and enjoying confluent-kafka-go without any issues, i can guarantee, that all the answers are in #128 and in some other GitHub issues. There is no "all use-cases" recipes available how to set this up, but it is totally doable and it is working pretty well. Just dig into GitHub issues.
In general you could take 2 paths. 1st - as suggested in #128 and 2nd - implement Linux Subsystem in VS Code (not sure how it plays with Goland).
P.S. As i remember there should also be some version match but do not remember which components.

**Comment by edenhill** [comment]
Closed in favour of #128
